interfaces/matlab_experimental/1D/Domain1D.m

```diff
@@ -0,0 +1,696 @@
+                        d.domainID = ctFunc('stflow_new', b.tpID, b.kinID, b.trID, 1);
+                    else
+                        error('Wrong argument type. Expecting instance of class Solution.');
+                    end
+
+                elseif strcmp(a, 'AxisymmetricFlow')
+
+                    if isa(b, 'Solution')
+                        d.domainID = ctFunc('stflow_new', b.tpID, b.kinID, b.trID, 2);
+                    else
+                        error('Wrong argument type. Expecting instance of class Solution.');
+                    end
+
+                elseif strcmp(a, 'ReactingSurface')
+
+                    if isa(b, 'Interface')
+                        d.domainID = ctFunc('reactingsurf_new');
+                        ctFunc('reactingsurf_setkineticsmgr', d.domainID, b.kinID);
+                    else
+                        error('Wrong argument type. Expecting instance of class Interface.');
+                    end
+
+                else
+                    error('Wrong object type.');
+                end
+
+            end
+
+            d.type = a;
+        end
+
+        %% Domain1D Class Destructor
+
+        function delete(d)
+            % Delete the :mat:class:`Domain1D` object.
+
+            ctFunc('domain_del', d.domainID);
+        end
+
+        %% Domain1D Utility Methods
+
+        function d = disableEnergy(d)
+            % Disable the energy equation. ::
+            %
+            %     >> d.disableEnergy
+            %
+            % :param d:
+            %     Instance of class :mat:class:`Domain1D`.
+
+            disp(' ');
+            disp('Disabling the energy equation...');
+            ctFunc('stflow_solveEnergyEqn', d.domainID, 0);
+        end
+
+        function d = enableEnergy(d)
+            % Enable the energy equation. ::
+            %
+            %     >> d.enableEnergy
+            %
+            % :param d:
+            %     Instance of class :mat:class:`Domain1D`.
+
+            disp(' ');
+            disp('Enabling the energy equation...');
+            ctFunc('stflow_solveEnergyEqn', d.domainID, 1);
+        end
+
+        function d = disableSoret(d)
+            % Disable the diffusive mass fluxes due to the Soret effect. ::
+            %
+            %     >> d.disableSoret
+            %
+            % :param d:
+            %     Instance of class :mat:class:`Domain1D`.
+
+            disp(' ');
+            disp('Disabling the Soret effect...');
+            ctFunc('stflow_enableSoret', d.domainID, 0);
+        end
+
+        function d = enableSoret(d)
+            % Enable the diffusive mass fluxes due to the Soret effect. ::
+            %
+            %     >> d.enableSoret
+            %
+            % :param d:
+            %     Instance of class :mat:class:`Domain1D`.
+
+            disp(' ');
+            disp('Disabling the Soret effect...');
+            ctFunc('stflow_enableSoret', d.domainID, 1);
+        end
```

Comment:
Should be handled by a property with getter/setter instead. The display should be eliminated.